src/mach/header.rs

```diff
@@ -180,6 +180,9 @@ pub struct Header32 {
 
 pub const SIZEOF_HEADER_32: usize = 0x1c;
 
+// # Safety
+//
+//   - `Header32` is exclusively made of `Plain` types (integers)
```

Comment:
Still `unsafe impl` instead of `#[macro_rules_derive(AsBytesAndFromBytes!)]`. Also a forgotten `git push`?

Reply:
Ideally the `::plain` crate would implement the pattern of this PR for its `Plain` type, so as to derive it without requiring `unsafe` (thanks to a `macro_rules!`).

Then I will be able to update this PR to use the new `::plain` and avoid requiring this semi-manually checked `unsafe impl Plain` (cc @le-jzr)

At which point there would not really be any `unsafe` left, except for the `unsafe fn`s and the `data_directories` indexing (which can be studied in a future PR) 🙂

Reply:
we can remove the data_directories indexing, i don't really care that much, unless someone else has some concerns